semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/Utils.java

```diff
@@ -0,0 +1,26 @@
+package org.hypertrace.semantic.convention.utils;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Splitter;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public class Utils {
+
+  private static final Splitter SLASH_SPLITTER = Splitter.on("/").omitEmptyStrings().trimResults();
+  private static final Joiner DOT_JOINER = Joiner.on(".");
+
+  /**
+   * Takes in a param path and converts it to a DOT joined path /service.product/GetProducts ->
+   * service.product.GetProducts
+   *
+   * @param path
+   * @return
+   */
+  public @Nullable static String sanitizePath(@Nonnull String path) {
```

Comment:
It feels like a no way back change in user data. Do you mind explaining a bit in the description of the method in which cases this method is being used?

Reply:
Sure. 

This is a utility function which was earlier present on the traceable side. This is moved here so that all methods accessing various http/grpc fields remain in one place.

This is being used to get http request header path in the httpsemantic file

Reply:
- Can we return an Optional, rather than null?
- Can we annotate the method, rather than inserting the `@Nullable` within the method signature?

Reply:
This code was already present in the traceable side and is just shifted here

Reply:
Sure. We can always improve the existing functions, if they are simple to refactor

Reply:
1. This is used over here so ig optional and returning null are equivalent over here

      return Optional.ofNullable(
              attributeValueMap.get(RPC_REQUEST_METADATA_PATH.getValue()).getValue())
          .map(path -> Utils.sanitizePath(path));

2. Not too sure about the @nullable notation

Reply:
A return type as Optional brings stronger typing into the system, than annotating a function with `@Nullable` annotation. The annotations are only used by the IDE to flag that things might need a null check. Using Optionals make things explicit

If the `sanitizePath` returns an Optional, the usage could just have used `flatMap` instead of `map`

```
return Optional.ofNullable(
        attributeValueMap.get(RawSpanConstants.getValue(HTTP_REQUEST_HEADER_PATH)).getValue())
    .flatMap(Utils::sanitizePath);
```

Reply:
Thanks for clarifying @sarthak77. I feel we need to explain the why of this method in the method comments, having here in the PR is good but none will look into the PR whenever they are in doubt of using/not using this method.

Reply:
@jcchavezs discussed on this with rishabh but we wanted to keep traceable and hypertrace separate as many folks might not know about the traceable platform

Reply:
> A return type as Optional brings stronger typing into the system, than annotating a function with `@Nullable` annotation. The annotations are only used by the IDE to flag that things might need a null check. Using Optionals make things explicit
> 
> If the `sanitizePath` returns an Optional, the usage could just have used `flatMap` instead of `map`
> 
> ```
> return Optional.ofNullable(
>         attributeValueMap.get(RawSpanConstants.getValue(HTTP_REQUEST_HEADER_PATH)).getValue())
>     .flatMap(Utils::sanitizePath);
> ```

Changes look good but the PR is merged. @skjindal93 should I open a separate PR for this?

Reply:
not urgent. You can take care of it, whenever you touch the same code later

Reply:
I need to modify this now only 😂
Since this is used as a submodule on traceable side some more changes are required here

Reply:
> @jcchavezs discussed on this with rishabh but we wanted to keep traceable and hypertrace separate as many folks might not know about the traceable platform

That is fine we don't need to mention traceable at all, my point is **there is a reason why we do this, we should be able to explain it in code, otherwise next committer might feel the need to remove it** and explanation can't be "because traceable does that".

Reply:
> > @jcchavezs discussed on this with rishabh but we wanted to keep traceable and hypertrace separate as many folks might not know about the traceable platform
> 
> That is fine we don't need to mention traceable at all, my point is **there is a reason why we do this, we should be able to explain it in code, otherwise next committer might feel the need to remove it** and explanation can't be "because traceable does that".

@jcchavezs ronak, rishabh and I had a long discussion on this method. This method was being used in traceable side and has no consumers on the HT side.

Maybe the purpose on traceable side is this (quoting @kotharironak )

" I think, by doing it that way learning and other component can compare the path irrespective of protocal "

Is this clear now ? (btw this method is moved back to traceable side)

Reply:
Hey @sarthak77 the point that @jcchavezs  is making is that if there is some piece of code added its purpose should be very clear, there is no need to bring into context about any specific consumer (traceable here). 
It is needed more so in this specific case since it was a public method, so there must be accompanying javadoc explaining the utility and intended usecase of the method, ow as jc mentioned it will pose maintenance challenges.

Reply:
Yes got it. Will keep it in mind.